Error while checking build/O2Physics/code-check for 059f628adeef3e37211e7152219fc8d6a53ffb7e at 2026-07-02 16:45:
```
## sw/BUILD/O2Physics-code-check-latest/log
--
========== List of issues found ==========
++ echo 'Found 28 errors and 323 warnings.'
Found 28 errors and 323 warnings.
++ [[ 28 -gt 0 ]]
++ cat /sw/BUILD/c4817d1f78f06c1c860c1488a68352f6fcf736e8/O2Physics-code-check/errors.txt
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:74:8: error: constructor does not initialize these fields: ccdb [cppcoreguidelines-pro-type-member-init,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:355:9: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:356:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:357:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:358:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:359:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:360:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:361:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:362:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1492:5: error: redundant return statement at the end of a function with a void return type [readability-redundant-control-flow,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1841:22: error: do not declare C-style arrays, use 'std::array' instead [modernize-avoid-c-arrays,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1860:22: error: do not declare C-style arrays, use 'std::array' instead [modernize-avoid-c-arrays,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:2186:7: error: do not use 'else' after 'return' [readability-else-after-return,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:3097:31: error: use 'contains' to check for membership [readability-container-contains,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphotonBkg.cxx:40:1: error: included header Vector4D.h is not used directly [misc-include-cleaner,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphotonBkg.cxx:65:8: error: constructor does not initialize these fields: ccdb [cppcoreguidelines-pro-type-member-init,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphotonBkg.cxx:499:7: error: do not use 'else' after 'return' [readability-else-after-return,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphotonBkg.cxx:593:21: error: no header providing "ROOT::Math::PtEtaPhiMVector" is directly included [misc-include-cleaner,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphotonBkg.cxx:706:23: error: no header providing "ROOT::Math::PxPyPzEVector" is directly included [misc-include-cleaner,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphoton.cxx:74:8: error: constructor does not initialize these fields: ccdb [cppcoreguidelines-pro-type-member-init,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphoton.cxx:245:9: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphoton.cxx:245:29: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphoton.cxx:679:5: error: redundant return statement at the end of a function with a void return type [readability-redundant-control-flow,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphoton.cxx:772:22: error: do not declare C-style arrays, use 'std::array' instead [modernize-avoid-c-arrays,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphoton.cxx:806:22: error: do not declare C-style arrays, use 'std::array' instead [modernize-avoid-c-arrays,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphoton.cxx:868:22: error: do not declare C-style arrays, use 'std::array' instead [modernize-avoid-c-arrays,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphoton.cxx:1017:22: error: do not declare C-style arrays, use 'std::array' instead [modernize-avoid-c-arrays,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphoton.cxx:1022:22: error: do not declare C-style arrays, use 'std::array' instead [modernize-avoid-c-arrays,-warnings-as-errors]
++ [[ 323 -gt 0 ]]
++ cat /sw/BUILD/c4817d1f78f06c1c860c1488a68352f6fcf736e8/O2Physics-code-check/warnings.txt
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:458:64: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:917:224: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:921:274: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1070:67: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1072:68: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1074:68: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1078:91: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1080:92: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1082:92: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1166:26: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1172:47: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1177:33: warning: statement should be inside braces [readability-braces-around-statements]
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/16904/059f628adeef3e37211e7152219fc8d6a53ffb7e/build_O2Physics_code-check/pretty.html).

--- a/PWGLF/TableProducer/Strangeness/sigma0builder.cxx
+++ b/PWGLF/TableProducer/Strangeness/sigma0builder.cxx
@@ -1374,11 +1374,11 @@ struct sigma0builder {
     if (eventSelections.maxIR >= 0 && interactionRate > eventSelections.maxIR) {
       return false;
     }
-    if (fillHists){
+    if (fillHists) {
       histos.fill(HIST("hEventSelection"), 19 /* Above max IR */);
       // Fill centrality histogram after event selection
       histos.fill(HIST("hEventCentrality"), centrality);
-    } 
+    }
     return true;
   }
 
@@ -2807,7 +2807,7 @@ struct sigma0builder {
           }
         }
 
-        if constexpr (!soa::is_table<TEMCal>) {// Don't use EMCal clusters here
+        if constexpr (!soa::is_table<TEMCal>) { // Don't use EMCal clusters here
           //_______________________________________________
           // KStar loop
           if (fillKStarTables) {

--- a/PWGLF/TableProducer/Strangeness/sigma0builder.cxx
+++ b/PWGLF/TableProducer/Strangeness/sigma0builder.cxx
@@ -981,9 +981,7 @@ struct sigma0builder {
         kShortMotherCode = MCMother_v02.pdgCode();
 
         // If the KShort mother is a (anti)Kaon, use the grandmother instead
-        if (std::abs(kShortMotherCode) == PDG_t::kK0 ||
-            std::abs(kShortMotherCode) == PDG_t::kK0Long ||
-            std::abs(kShortMotherCode) == PDG_t::kKPlus) {
+        if (std::abs(kShortMotherCode) == PDG_t::kK0) {
           auto const& kShortGrandMothers = MCMother_v02.template mothers_as<aod::McParticles>();
           if (!kShortGrandMothers.empty()) {
             kShortMotherCode = kShortGrandMothers.front().pdgCode();
@@ -1376,13 +1374,11 @@ struct sigma0builder {
     if (eventSelections.maxIR >= 0 && interactionRate > eventSelections.maxIR) {
       return false;
     }
-    if (fillHists)
+    if (fillHists){
       histos.fill(HIST("hEventSelection"), 19 /* Above max IR */);
-
-    // Fill centrality histogram after event selection
-    if (fillHists)
+      // Fill centrality histogram after event selection
       histos.fill(HIST("hEventCentrality"), centrality);
-
+    } 
     return true;
   }
 
@@ -2811,9 +2807,9 @@ struct sigma0builder {
           }
         }
 
-        //_______________________________________________
-        // KStar loop
-        if constexpr (!soa::is_table<TEMCal>) { // Don't use EMCal clusters here
+        if constexpr (!soa::is_table<TEMCal>) {// Don't use EMCal clusters here
+          //_______________________________________________
+          // KStar loop
           if (fillKStarTables) {
             auto gamma1 = fullV0s.rawIteratorAt(bestGammasArray[i]);
             for (size_t j = 0; j < bestKShortsArray.size(); ++j) {
@@ -2824,11 +2820,9 @@ struct sigma0builder {
                 continue;
             }
           }
-        }
 
-        //_______________________________________________
-        // pi0 loop
-        if constexpr (!soa::is_table<TEMCal>) { // Don't use EMCal clusters here
+          //_______________________________________________
+          // pi0 loop
           if (fillPi0Tables) {
             auto gamma1 = fullV0s.rawIteratorAt(bestGammasArray[i]);
             for (size_t j = i + 1; j < bestGammasArray.size(); ++j) {

--- a/PWGLF/Tasks/Resonances/k892hadronphoton.cxx
+++ b/PWGLF/Tasks/Resonances/k892hadronphoton.cxx
@@ -579,13 +579,13 @@ struct k892hadronphoton {
     if (eventSelections.maxIR >= 0 && interactionRate > eventSelections.maxIR) {
       return false;
     }
-    if (fillHists)
+
+    if (fillHists){
       histos.fill(HIST("hEventSelection"), 19 /* Above max IR */);
-
-    // Fill centrality histogram after event selection
-    if (fillHists)
+      // Fill centrality histogram after event selection
       histos.fill(HIST("hEventCentrality"), centrality);
-
+    }
+    
     return true;
   }
 

--- a/PWGLF/Tasks/Resonances/k892hadronphoton.cxx
+++ b/PWGLF/Tasks/Resonances/k892hadronphoton.cxx
@@ -580,12 +580,12 @@ struct k892hadronphoton {
       return false;
     }
 
-    if (fillHists){
+    if (fillHists) {
       histos.fill(HIST("hEventSelection"), 19 /* Above max IR */);
       // Fill centrality histogram after event selection
       histos.fill(HIST("hEventCentrality"), centrality);
     }
-    
+
     return true;
   }
 

--- a/PWGLF/Tasks/Resonances/k892hadronphotonBkg.cxx
+++ b/PWGLF/Tasks/Resonances/k892hadronphotonBkg.cxx
@@ -386,7 +386,7 @@ struct k892hadronphotonBkg {
     if (eventSelections.maxIR >= 0 && interactionRate > eventSelections.maxIR) {
       return false;
     }
-    if (fillHists){
+    if (fillHists) {
       histos.fill(HIST("hEventSelection"), 19 /* Above max IR */);
       // Fill centrality histogram after event selection
       histos.fill(HIST("hEventCentrality"), centrality);

--- a/PWGLF/Tasks/Resonances/k892hadronphotonBkg.cxx
+++ b/PWGLF/Tasks/Resonances/k892hadronphotonBkg.cxx
@@ -386,13 +386,11 @@ struct k892hadronphotonBkg {
     if (eventSelections.maxIR >= 0 && interactionRate > eventSelections.maxIR) {
       return false;
     }
-    if (fillHists)
+    if (fillHists){
       histos.fill(HIST("hEventSelection"), 19 /* Above max IR */);
-
-    // Fill centrality histogram after event selection
-    if (fillHists)
+      // Fill centrality histogram after event selection
       histos.fill(HIST("hEventCentrality"), centrality);
-
+    }
     return true;
   }
 

